x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -134,92 +134,41 @@ class AgentPolicyService {
       is_preconfigured: true,
     };
 
-    let searchParams;
-
-    const isDefaultPolicy =
-      preconfiguredAgentPolicy.is_default || preconfiguredAgentPolicy.is_default_fleet_server;
-
-    if (isDefaultPolicy) {
-      searchParams = {
-        searchFields: [
-          preconfiguredAgentPolicy.is_default_fleet_server
-            ? 'is_default_fleet_server'
-            : 'is_default',
-        ],
-        search: 'true',
-      };
-    } else if (id) {
-      searchParams = {
-        id: String(id),
-      };
-    }
-
-    if (!searchParams) throw new Error('Missing ID');
+    if (!id) throw new Error('Missing ID');
 
-    return await this.ensureAgentPolicy(soClient, esClient, newAgentPolicy, searchParams, id);
+    return await this.ensureAgentPolicy(soClient, esClient, newAgentPolicy, id as string);
   }
 
   private async ensureAgentPolicy(
     soClient: SavedObjectsClientContract,
     esClient: ElasticsearchClient,
     newAgentPolicy: NewAgentPolicy,
-    searchParams:
-      | { id: string }
-      | {
-          searchFields: string[];
-          search: string;
-        },
-    id?: string | number
+    id: string
   ): Promise<{
     created: boolean;
     policy: AgentPolicy;
   }> {
     // For preconfigured policies with a specified ID
-    if ('id' in searchParams) {
-      try {
-        const agentPolicy = await soClient.get<AgentPolicySOAttributes>(
-          AGENT_POLICY_SAVED_OBJECT_TYPE,
-          searchParams.id
-        );
-        return {
-          created: false,
-          policy: {
-            id: agentPolicy.id,
-            ...agentPolicy.attributes,
-          },
-        };
-      } catch (e) {
-        if (SavedObjectsErrorHelpers.isNotFoundError(e)) {
-          return {
-            created: true,
-            policy: await this.create(soClient, esClient, newAgentPolicy, { id: searchParams.id }),
-          };
-        } else throw e;
-      }
-    }
-
-    // For default policies without a specified ID
-    const agentPolicies = await soClient.find<AgentPolicySOAttributes>({
-      type: AGENT_POLICY_SAVED_OBJECT_TYPE,
-      ...searchParams,
-    });
-
-    if (agentPolicies.total === 0) {
+    try {
+      const agentPolicy = await soClient.get<AgentPolicySOAttributes>(
+        AGENT_POLICY_SAVED_OBJECT_TYPE,
+        id
+      );
       return {
-        created: true,
-        policy: await this.create(soClient, esClient, newAgentPolicy, {
-          id: id ? String(id) : uuidv5(newAgentPolicy.name, UUID_V5_NAMESPACE),
-        }),
+        created: false,
+        policy: {
+          id: agentPolicy.id,
+          ...agentPolicy.attributes,
+        },
       };
+    } catch (e) {
```

Comment:
nit: let's not use exceptions for normal control flow. Let's change the `get` call above to fallback to undefined and then do a simple if/else.

Reply:
`soClient.get` is coming from `src/core/server/saved_objects`, I'm not keen on changing that, it might have a large impact. let me create a helper function instead.

Reply:
done